src/theory/strings/theory_strings.cpp

```diff
@@ -2077,6 +2077,20 @@ void TheoryStrings::checkExtfInference( Node n, Node nr, ExtfInfoTmp& in, int ef
             do_infer = pol ? !areEqual(lit[0], lit[1])
                            : !areDisequal(lit[0], lit[1]);
           }
+          else if (lit.getKind() == OR)
+          {
+            do_infer = pol;
```

Comment:
I don't quite get it. Let's say that we have `(not (or (str.contains x y) (str.contains z w)))` and both `(str.contains x y)` and `(str.contains z w)` are not equal to true in the current context. Don't we want to do `CTN_Trans` in that case?